examples/bodyparser.rs

```diff
@@ -1,31 +1,50 @@
-#![feature(globs)]
 extern crate iron;
 extern crate bodyparser;
-extern crate serialize;
+extern crate persistent;
+extern crate "rustc-serialize" as rustc_serialize;
 
-use iron::prelude::*;
+use persistent::Read;
 use iron::status;
-use iron::response::modifiers::{Body, Status};
-use bodyparser::BodyParser;
+use iron::prelude::*;
 
-#[deriving(Decodable, Clone, Show)]
-struct JsonParams {
-    name: String,
-    age: Option<i8>,
+#[derive(Debug, Clone, RustcDecodable)]
+struct MyStructure {
+    a: String,
+    b: Option<String>,
 }
 
-fn log_json(req: &mut Request) -> IronResult<Response> {
-    let parsed = req.get::<BodyParser<JsonParams>>();
-    match parsed {
-        Some(params) => println!("Parsed json:\n{}", params),
-        None => println!("Invalid or no json!"),
+fn log_body(req: &mut Request) -> IronResult<Response> {
+    let body = req.get::<bodyparser::BodyReader>();
+    match body {
+        Ok(Some(body)) => println!("Read body:\n{}", body),
+        Ok(None) => println!("No body"),
+        Err(err) => println!("Error: {:?}", err)
+    }
+
+    let json_body = req.get::<bodyparser::JsonBodyParser>();
```

Comment:
I think this is a bit too verbose. Could you remove the `BodyParser` suffix from all these types and rename `BodyReader` to `Raw`?